test/e2e/task_start_e2e_test.go

```diff
@@ -0,0 +1,147 @@
+// Copyright © 2020 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	tb "github.com/tektoncd/pipeline/test/builder"
+	"gotest.tools/v3/assert"
+	is "gotest.tools/v3/assert/cmp"
+	"gotest.tools/v3/icmd"
+	knativetest "knative.dev/pkg/test"
+)
+
+func TestTaskStartE2E(t *testing.T) {
+
+	t.Parallel()
+	c, namespace := Setup(t, WithRegistry)
+	knativetest.CleanupOnInterrupt(func() { TearDown(t, c, namespace) }, t.Logf)
+	defer TearDown(t, c, namespace)
+
+	repo := fmt.Sprintf("registry.%s:5000/buildahtest", namespace)
+
+	run := TestEnv.Prepare(t)
+
+	t.Run("Create buildah task in namespace "+namespace, func(t *testing.T) {
+		res := icmd.RunCmd(icmd.Cmd{Command: []string{"kubectl", "apply", "-f", TestResourcePath("buildah.yaml"), "-n", namespace}, Timeout: timeout})
+		res.Assert(t, icmd.Expected{
+			ExitCode: 0,
+			Err:      icmd.None,
+		})
+		time.Sleep(1 * time.Second)
+	})
+
+	t.Run("Get list of Tasks from namespace  "+namespace, func(t *testing.T) {
+		res := icmd.RunCmd(run("task", "list", "-n", namespace))
+
+		expected := ListAllTasksOutput(t, c, map[int]interface{}{
+			0: &TaskData{
+				Name: "buildah",
+			},
+		})
+
+		res.Assert(t, icmd.Expected{
+			ExitCode: 0,
+			Err:      icmd.None,
+			Out:      expected,
+		})
+	})
+
+	t.Run("create buildah resources in namespace "+namespace, func(t *testing.T) {
```

Comment:
Could this not be part of the set up of another test? Why is this it's own test?